client/state/reader/follows/reducer.js

```diff
@@ -11,11 +11,95 @@ import {
 	READER_FOLLOW,
 	READER_UNFOLLOW,
 	READER_FOLLOWS_RECEIVE,
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL,
 } from 'state/action-types';
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
 /**
+ * Update a post subscription
+ *
+ * @param {object} state The current state from the items reducer
+ * @param {object} payload the payload from the subscribe / update / unsubscribe action
+ * @param {function} isEqualToCurrentState A function that determines if the payload will actually change the current state.
+ * Returns true if the state would not change.
+ * @param {object} newProps The new props to set on the delivery_methods.email object if the state will change
+ * @returns {object} the next state
+ */
+function updatePostSubscription( state, payload, isEqualToCurrentState, newProps ) {
```

Comment:
why does `newProps` exist? Isn't all that info within the action type + payload?